core/src/main/java/io/cloudevents/core/v03/CloudEventBuilder.java

```diff
@@ -50,6 +50,20 @@ public CloudEventBuilder(io.cloudevents.CloudEvent event) {
         super(event);
     }
 
+    public CloudEventBuilder(io.cloudevents.CloudEventContext context) {
+        super();
+        for (String name: context.getAttributeNames()) {
+            if (!name.equals(CloudEventV03.SPECVERSION)) {
+                Object value = context.getAttribute(name);
+                if (value instanceof String) {
+                    withAttribute(name, (String) value);
+                } else {
+                    withAttribute(name, value.toString());
+                }
+            }
+        }
+    }
```

Comment:
Could this live in the base builder?

Reply:
I guess it could. But you couldn't use the static constant in that case. If that's OK with you I can make that change (or you can).